src/woo_publications/contrib/documents_api/client.py

```diff
@@ -149,8 +158,19 @@ def create_document(
         )
 
     def destroy_document(self, uuid: UUID) -> None:
-        response = self.delete(f"enkelvoudiginformatieobjecten/{uuid}")
-        response.raise_for_status()
+        try:
+            response = self.delete(f"enkelvoudiginformatieobjecten/{uuid}")
+            response.raise_for_status()
+        except RequestException as err:
+            if (
+                hasattr(err.response, "status_code") and err.response.status_code == 404  # pyright: ignore[reportOptionalMemberAccess]
+            ):
+                return
```

Comment:
requests ensures that you get a `Response` instance, so you don't need to do dynamic lookups

```suggestion
            if (err.response is not None and err.response.status_code == 404):
                return
```
the type checker should then also be able to narrow the type and not complain about optional member access